### PR TITLE
[#27] Migrate to JUnit 5: org.operaton.connect submodules

### PR DESCRIPTION
**Pull request by kthoms**
Migrate org.operaton.connect submodules to JUnit 5.

Performed Open Rewrite rule org.openrewrite.java.testing.junit5.JUnit5BestPractices and manually completed the migration.

Upgrade assertj to 3.26.3.

### CODE REVIEW

**Comment by kthoms** [comment]
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project operaton-dmn-junit5: Compilation failure: Compilation failure: 
Error:  /home/runner/work/operaton/operaton/test-utils/junit5-extension-dmn/src/main/java/org/operaton/bpm/dmn/engine/test/junit5/DmnEngineExtension.java:[26,39] package org.junit.jupiter.api.extension does not exist
```

File: engine-dmn/engine/src/test/java/org/operaton/bpm/dmn/engine/test/asserts/DmnDecisionTableResultAssert.java [hunk 1]
```diff
@@ -40,4 +43,12 @@ protected DmnDecisionRuleResultAssert toAssert(DmnDecisionRuleResult value, Stri
 
     return new DmnDecisionRuleResultAssert(value);
   }
+
```
Comment (kthoms) [inline engine-dmn/engine/src/test/java/org/operaton/bpm/dmn/engine/test/asserts/DmnDecisionTableResultAssert.java:46]: **Review Hint**: New required method by assertj upgrade

File: engine/src/test/java/org/operaton/bpm/engine/test/api/history/removaltime/batch/BatchSetRemovalTimeTest.java [hunk 2]
```diff
@@ -1589,7 +1590,7 @@ public void shouldSetRemovalTimeForStandaloneDecision_Null() {
       .singleResult();
 
     // then
-    assertThat(historicDecisionInstance.getRemovalTime()).isEqualTo(null);
```
Comment (kthoms) [inline engine/src/test/java/org/operaton/bpm/engine/test/api/history/removaltime/batch/BatchSetRemovalTimeTest.java:1592]: **Review Hint**: expression became ambiguous by assertj upgrade

File: engine/src/test/java/org/operaton/bpm/engine/test/concurrency/AbstractCompetingTransactionsOptimisticLockingTest.java [hunk 3]
```diff
@@ -124,7 +125,7 @@ public void shouldTreatConcurrentDeletionOfExecutionForTaskInsertAsForeignKeyExc
     assertThat(thread1.exception)
       .isInstanceOf(ProcessEngineException.class)
       .extracting("code")
-      .contains(BuiltinExceptionCode.FOREIGN_KEY_CONSTRAINT_VIOLATION.getCode());
+      .isEqualTo(BuiltinExceptionCode.FOREIGN_KEY_CONSTRAINT_VIOLATION.getCode());
```
Comment (kthoms) [inline engine/src/test/java/org/operaton/bpm/engine/test/concurrency/AbstractCompetingTransactionsOptimisticLockingTest.java:128]: **Review Hint**: contains not available after assertj upgrade

File: engine/src/test/java/org/operaton/bpm/engine/test/standalone/scripting/EnvScriptResolutionTest.java [hunk 4]
```diff
@@ -58,8 +61,8 @@ public void shouldFindEnvScriptForLanguage() {
     assertThat(environmentScripts)
       .hasSize(1)
       .containsKey(ECMASCRIPT_LANGUAGE)
-      .extracting(ECMASCRIPT_LANGUAGE)
-        .hasSize(1);
+      .extracting(ECMASCRIPT_LANGUAGE, as(InstanceOfAssertFactories.list(Object.class)))
```
Comment (kthoms) [inline engine/src/test/java/org/operaton/bpm/engine/test/standalone/scripting/EnvScriptResolutionTest.java:64]: **Review Hint**: required after assertj upgrade

**Comment by javahippie** [comment]
lgtm, but has a conflict on the branch

**Pull request merged**